lib/everypolitician/popolo/collection.rb

```diff
@@ -35,11 +37,21 @@ def where(attributes = {})
         attributes.map { |k, v| index_for(k.to_sym)[v].to_a }.reduce(:&) || []
       end
 
+      def of_class(klass)
+        @of_class[klass] ||= select { |e| e.class == klass }
+      end
+
       private
 
       def index_for(attr)
         @indexes[attr] ||= group_by(&attr)
       end
+
+      def class_for_entity(document)
+        @event_class[document[:classification]] ||= self.class.entity_class.subclasses.select do |e|
```

Comment:
I think this can be a `find` instead of `select`, then it will just return the first match and you can drop the `.first` on line 53.